alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteMonitorAction.kt

```diff
@@ -130,7 +130,7 @@ class TransportExecuteMonitorAction @Inject constructor(
                                 docLevelMonitorQueries.initDocLevelQueryIndex(monitor.dataSources)
                                 log.info("Central Percolation index ${ScheduledJob.DOC_LEVEL_QUERIES_INDEX} created")
                             }
-                            val (metadata, _) = MonitorMetadataService.getOrCreateMetadata(monitor)
+                            val (metadata, _) = MonitorMetadataService.getOrCreateMetadata(monitor, skipIndex = true)
```

Comment:
This should only be true if the dryrun param is true

Reply:
even if monitorId is null ?

Reply:
I think if monitorId is null we definitely can't save metadata (there's no monitor to tie it to). 

Dryrun can be true for existing monitor and we take that into account before indexing metadata inside DocumentLevelMonitorRunner

Reply:
Oh I missed that part of the logic, good call out.